ci/build_wheel_libcuspatial.sh

```diff
@@ -3,4 +3,34 @@
 
 set -euo pipefail
 
-ci/build_wheel.sh libcuspatial python/libcuspatial cpp
+package_name="libcuspatial"
+
+git clone \
+  --branch multiple-file-keys \
+  https://github.com/jameslamb/dependency-file-generator.git \
+  /tmp/rapids-dependency-file-generator
+
+pip uninstall --yes rapids-dependency-file-generator
+pip install /tmp/rapids-dependency-file-generator
+
+rapids-logger "Generating build requirements"
+matrix_selectors="cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION};cuda_suffixed=true"
+
+rapids-dependency-file-generator \
+  --output requirements \
+  --file-key "py_build_${package_name}" \
+  --file-key "py_rapids_build_${package_name}" \
+  --matrix "${matrix_selectors}" \
+| tee /tmp/requirements-build.txt
+
+rapids-logger "Installing build requirements"
+python -m pip install \
+    -v \
+    --prefer-binary \
+    -r /tmp/requirements-build.txt
+
+# build with '--no-build-isolation', for better sccache hit rate
+# 0 really means "add --no-build-isolation" (ref: https://github.com/pypa/pip/issues/5735)
+export PIP_NO_BUILD_ISOLATION=0
```

Comment:
Ok I think this is working.

On the latest build on `branch-24.12`, you can see log lines about installing build dependencies and creating a temporary build directory.

```text
Created temporary directory: /tmp/pip-wheel-ndbc3klu
Created temporary directory: /tmp/pip-ephem-wheel-cache-blfbvohw
Looking in indexes: https://pypi.org/simple, https://pypi.anaconda.org/rapidsai-wheels-nightly/simple, https://pypi.nvidia.com/
Processing /__w/cuspatial/cuspatial/python/libcuspatial
  Added file:///__w/cuspatial/cuspatial/python/libcuspatial to build tracker '/tmp/pip-build-tracker-vn6jabrn'
  Created temporary directory: /tmp/pip-build-env-64dc7d88
  Running command pip subprocess to install build dependencies
  Installing build dependencies: started
```

([build link](https://github.com/rapidsai/cuspatial/actions/runs/11378013030/job/31653081303#step:9:19))

On this PR, none of that.

```text
Looking in indexes: https://pypi.org/simple, https://pypi.anaconda.org/rapidsai-wheels-nightly/simple, https://pypi.nvidia.com/
Processing /__w/cuspatial/cuspatial/python/libcuspatial
  Running command Preparing metadata (pyproject.toml)
  Preparing metadata (pyproject.toml): started
```

([build link](https://github.com/rapidsai/cuspatial/actions/runs/11390386364/job/31691951451?pr=1473#step:9:133))

Reply:
I mention that because it looks to me like we're not getting `sccache` hits.

```text
Successfully built libcuspatial-cu11
Compile requests                      28
Compile requests executed             28
Cache hits                             4
Cache hits (c [gcc])                   2
```

There hasn't been a new `rmm` commit merged since yesterday, and the last new `cudf` commit merged was an hour ago.

Gonna try another build to double-check this.

Reply:
Yeah cool ok, I just did another build and saw all cache hits. Must have just gotten unlucky before with new `cudf` changes being committed and new nightlies appearing.

```text
Compile requests                     28
Compile requests executed            28
Cache hits                           28
Cache hits (c [gcc])                  2
Cache hits (c++ [gcc])                2
Cache hits (cuda [nvcc])             24
```

([build link](https://github.com/rapidsai/cuspatial/actions/runs/11391151277/job/31694320694?pr=1473))

Reply:
Awesome thanks for being careful with the checking.